discovery_engine_core/web-api2/src/ingestion.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::net::SocketAddr;
+
+use actix_web::{
+    web::{self, Data, Json, ServiceConfig},
+    Responder,
+};
+use serde::Deserialize;
+
+use crate::{
+    config::Config,
+    error::application::{Unimplemented, WithRequestIdExt},
+    server::{default_bind_address, Application},
+    Error,
+};
+
+pub struct Ingestion;
+
+impl Application for Ingestion {
+    type Config = IngestionConfig;
+    type AppState = IngestionConfig;
+
+    fn configure(config: &mut ServiceConfig) {
+        let resource = web::resource("/documents")
+            .route(web::post().to(new_documents.error_with_request_id()));
+        config.service(resource);
+    }
+}
+
+#[derive(Deserialize)]
+pub struct IngestionConfig {
+    #[serde(default = "default_bind_address")]
+    bind_to: SocketAddr,
+}
+
+impl Config for IngestionConfig {
+    fn bind_address(&self) -> std::net::SocketAddr {
+        self.bind_to
+    }
+}
+
+//FIXME use actual body
+#[derive(Deserialize)]
+struct NewDocuments {}
```

Comment:
maybe we could use models from original `web-api` crate instead as a way to have some "continuation" in the code?

Reply:
yes, but in follow up PRs